### ISSUE TITLE
False positive: deprecated-bare-vars on relative path to with_filetree

### ISSUE THREAD

**Issue by bluikko** [ansible/ansible-lint#1621]
##### Summary

Giving a relative path that does not end with a slash to `with_filetree` causes false positive for `deprecated-bare-vars`.
```console
deprecated-bare-vars: Found a bare variable 'templates/files.d' used in a 'with_filetree' loop. You should use the full variable syntax ('{{ templates/files.d }}')
```

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
ansible --version
ansible 2.9.22
  python version = 3.6.8 (default, Nov 16 2020, 16:55:22) [GCC 4.8.5 20150623 (Red Hat 4.8.5-44)]
ansible-lint --version
ansible-lint 5.0.11 using ansible 2.9.22
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT

EL7.9 fully updated

##### STEPS TO REPRODUCE

```yaml
- name: Copy files
  template:
    src: '{{ item.src }}'
    dest: "/tmp/files.d/{{ item.path | regex_replace('.j2') }}"
    mode: '0644'
  with_filetree: templates/files.d
  when: item.state == 'file'
```

##### Desired Behaviour

It should not complain since this is not a variable it is a path.

Suggest that if `with_filetree` contains slashes (possibly without spaces around the slash? if it is being confused with number division) then it is not treated as a bare var.

##### Actual Behaviour

The relative path is detected as a bare variable.

A fix is to add a slash at the end of the path, for example:
```yaml
- name: Copy files
  template:
    src: '{{ item.src }}'
    dest: "/tmp/files.d/{{ item.path | regex_replace('.j2') }}"
    mode: '0644'
  with_filetree: templates/files.d/
  when: item.state == 'file'
```
This makes ansible-lint not detect the "bare var". But usually paths do not contain the final slash at the end.

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by philipsd6** [comment]
> A fix is to add a slash at the end of the path

I find this is not true in my case, as when I run ansible-lint 5.3.1 I see this:

    playbook.yaml:42: deprecated-bare-vars Found a bare variable 'templates/mydir/' used in a 
    'with_community.general.filetree' loop. You should use the full variable syntax ('{{ templates/mydir/ }}')

The task looks like this:

```yaml
    - name: Create my directories
      file:
        path: "{{ temp.path }}/{{ item.path }}"
        state: directory
        mode: "{{ item.mode }}"
      with_community.general.filetree: templates/mydir/
      when: item.state == "directory"
```

**Comment by bluikko** [comment]
> I find this is not true in my case, as when I run ansible-lint 5.3.1

It may be, I had already changed all `with_` loops to just plain `loop:` before 5.3.1.

**Comment by philipsd6** [comment]
> generator object unrolling is automatic

Nice! I hadn't noticed that in any release notes, but I _try_ to write code that people can run with the oldest supported releases anyway, even if I always develop right off the `devel` branch.

**Comment by lr1980** [comment]
`with_community.general.filetree: "templates/{{ my_var }}/"`
is also a false positive

**Comment by mdebord-dlr** [comment]
This also throws: `with_file: "~/myKey.gpg"` 

```
Error: deprecated-bare-vars Found a bare variable '~/myKey.gpg' used in a 'with_file' loop. You should use the full variable syntax ('{{ ~/myKey.gpg }}')
```

**Comment by mdebord-dlr** [comment]
For anyone else hitting this, I got this error with:
```
  with_file: "~/key.gpg"
```

This did not flag the lint error: 

```
  with_file:
    - "~/key.gpg"
```

**Comment by ssbarnea** [comment]
Should I also mention that with_flattened was removed in ansible 2.13, so it does not make sense to complain about it.

**Comment by olibre** [comment]
Yep @ssbarnea  Just successfully tested with `"{{ 'templates/' }}"`

```yaml
- name: Create remote directories
  ansible.builtin.file:
    path: /remote/path/{{ item.path }}
    state: directory
    mode: a=,u=rX
  with_community.general.filetree: "{{ 'templates/' }}"
  when: item.state == "directory"
```
